### PR TITLE
Feature. Schema test agent(#19)

### LINKED ISSUE #19: Database: create schema for tests and agents
Create and support schema for tests and agent status in save-common. Schema should use JPA-API. 

Schema for tests should cover test run status.
Schema for agents should cover agent status.

### PR DESCRIPTION
**Pull request by aktsay6**
### What's done:
  * Made schema for agent

This pull request closes #19

### CODE REVIEW

File: save-common/src/jvmMain/kotlin/org/cqfn/save/entities/AgentStatus.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package org.cqfn.save.entities
+
+import org.cqfn.save.agent.AgentState
+import java.sql.Date
+import javax.persistence.*
+
+@Entity
+class AgentStatus (
+        @Column(name="time")
+        var time: Date,
```
Comment (petertrr) [inline save-common/src/jvmMain/kotlin/org/cqfn/save/entities/AgentStatus.kt:10]: I believe it should be LocalDate

File: save-common/src/jvmMain/kotlin/org/cqfn/save/entities/AgentStatus.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package org.cqfn.save.entities
+
+import org.cqfn.save.agent.AgentState
+import java.sql.Date
+import javax.persistence.*
+
+@Entity
+class AgentStatus (
```
Comment (petertrr) [inline save-common/src/jvmMain/kotlin/org/cqfn/save/entities/AgentStatus.kt:8]: There should also be a field for agent id or container id, not only generated id for DB

File: save-common/src/jvmMain/kotlin/org/cqfn/save/entities/AgentStatus.kt (hunk 1 above)
Comment (aktsay6) [reply save-common/src/jvmMain/kotlin/org/cqfn/save/entities/AgentStatus.kt]: Fixed

File: save-common/src/jvmMain/kotlin/org/cqfn/save/entities/AgentStatus.kt (hunk 2 above)
Comment (aktsay6) [reply save-common/src/jvmMain/kotlin/org/cqfn/save/entities/AgentStatus.kt:8]: Fixed

File: save-common/src/jvmMain/kotlin/org/cqfn/save/entities/AgentStatus.kt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import java.time.LocalDate
+import javax.persistence.*
+
+@Entity
+class AgentStatus (
+        @Column(name="time")
+        var time: LocalDate,
+        @Enumerated(EnumType.STRING)
+        var state: AgentState,
+        var agentId: Long,
```
Comment (petertrr) [inline save-common/src/jvmMain/kotlin/org/cqfn/save/entities/AgentStatus.kt:13]: Agent id, or container ID if we use it too, comes from UUID and is a string

File: buildSrc/src/main/kotlin/org/cqfn/save/buildutils/SpringBootConfiguration.kt [hunk 4]
```diff
@@ -26,16 +24,6 @@ fun Project.configureSpringBoot(withSpringDataJpa: Boolean = false) {
         buildInfo()  // configures `bootBuildInfo` task, which creates META-INF/build-info.properties file
     }
 
-    if (withSpringDataJpa) {
-        apply<AllOpenGradleSubplugin>()
```
Comment (petertrr) [inline buildSrc/src/main/kotlin/org/cqfn/save/buildutils/SpringBootConfiguration.kt:30]: You can add dependencies here: data-jpa, h2, reflect

File: save-common/src/jvmMain/kotlin/org/cqfn/save/entities/AgentStatus.kt [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package org.cqfn.save.entities
```
Comment (petertrr) [inline save-common/src/jvmMain/kotlin/org/cqfn/save/entities/AgentStatus.kt:1]: Please also add to readme in save-common, what we have discussed about common sources: entities can be placed in jvmMain if they are not intended to be used on frontend, and if usage of platform-specific APIs becomes to complex, we'll need to create a corresponding common class (which will be serizlizable) for the entity

**Review by petertrr — CHANGES_REQUESTED** [review]

File: save-common/src/jvmMain/kotlin/org/cqfn/save/entities/AgentStatus.kt [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+import org.cqfn.save.agent.AgentState
+import java.time.LocalDate
```
Comment (akuleshov7) [inline save-common/src/jvmMain/kotlin/org/cqfn/save/entities/AgentStatus.kt:4]: just a remark to @aktsay6: https://docs.liquibase.com/change-types/community/add-unique-constraint.html
this table should have unique constraints. Keep this in mind

File: save-common/src/jvmMain/kotlin/org/cqfn/save/entities/AgentStatus.kt (hunk 3 above)
Comment (akuleshov7) [inline save-common/src/jvmMain/kotlin/org/cqfn/save/entities/AgentStatus.kt:13]: agentId should be unique: https://docs.jboss.org/hibernate/jpa/2.1/api/javax/persistence/UniqueConstraint.html

File: save-orchestrator/src/main/resources/data.sql [hunk 7]
```diff
@@ -0,0 +1,13 @@
+DROP TABLE IF EXISTS agent_status;
```
Comment (akuleshov7) [inline save-orchestrator/src/main/resources/data.sql:1]: as we have decided - these scripts should be added to common gradle liquibase script

File: save-orchestrator/src/main/resources/data.sql [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
+CREATE TABLE agent_status (
+                         id INT AUTO_INCREMENT  PRIMARY KEY,
```
Comment (akuleshov7) [inline save-orchestrator/src/main/resources/data.sql:4]: Bigint

**Comment by kentr0w** [comment]
Add `project_id`, `test_suite_execution` columns in `test_status` table. And also add `test_suite_execution` table.

File: save-common/src/jvmMain/kotlin/org/cqfn/save/entities/TestStatus.kt [hunk 9]
```diff
@@ -0,0 +1,12 @@
+package org.cqfn.save.entities
+
+import java.time.LocalDateTime
+
+@Entity
+class TestStatus (
+        var status: String,
```
Comment (petertrr) [inline save-common/src/jvmMain/kotlin/org/cqfn/save/entities/TestStatus.kt:7]: Should be a enum of values, as stated in the isseu

**Comment by petertrr** [comment]
> Add `project_id`, `test_suite_execution` columns in `test_status` table. And also add `test_suite_execution` table.

`test_suite_execution` can be added in a separate PR, this PR focuses on statuses of `test_execution`s and agent statues

**Review by petertrr — CHANGES_REQUESTED** [review]

File: save-orchestrator/src/main/resources/application-dev.properties [hunk 10]
```diff
@@ -0,0 +1,14 @@
+spring.h2.console.enabled=true
```
[the remark this answers is not in the document]
Comment (aktsay6) [reply save-orchestrator/src/main/resources/application-dev.properties]: Yes, deleted

**Review by petertrr — APPROVED** [review]
lgtm

**Pull request merged**